### PR TITLE
Fix exception parameter not being used

### LINKED ISSUE #193: Unused exception variable warning not detected by CI
The exception parameter `ex` is not used and should have caused CI to fail!

_Originally posted by @achim-k in https://github.com/foxglove/ros-foxglove-bridge/pull/191#discussion_r1132433845_

### PR DESCRIPTION
**Pull request by achim-k**
### Public-Facing Changes
None

### Description
- Enables stricter clang compilation flags
- Fix unused exception parameter

fixes #193

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -17,7 +17,10 @@ set(CMAKE_CXX_STANDARD_REQUIRED ON)
 macro(enable_strict_compiler_warnings target)
   target_compile_options(${target} PRIVATE
     $<$<CXX_COMPILER_ID:MSVC>:/WX /W4>
-    $<$<NOT:$<CXX_COMPILER_ID:MSVC>>:-Wall -Wextra -Wpedantic -Werror -Wold-style-cast -Wfloat-equal>
```
Comment (jtbandes) [inline CMakeLists.txt:20]: This conditional stuff is crazy – can we not just use regular `if()` to do this?

**Review by jtbandes — APPROVED** [review]
Personally, my strategy with clang is usually to start with `-Weverything` and disable the annoying warnings.

File: CMakeLists.txt (hunk 1 above)
Comment (achim-k) [reply CMakeLists.txt:20]: I can change it to that, yes

**Comment by achim-k** [comment]
> Personally, my strategy with clang is usually to start with `-Weverything` and disable the annoying warnings.

Had this at the beginning, but raised to many compile warnings. We should have done this from the start

**Pull request merged**